flang/lib/Optimizer/CodeGen/TypeConverter.h

```diff
@@ -127,8 +127,13 @@ class LLVMTypeConverter : public mlir::LLVMTypeConverter {
   // i64 can be used to index into aggregates like arrays
   mlir::Type indexType() { return mlir::IntegerType::get(&getContext(), 64); }
 
-  // TODO
-  bool requiresExtendedDesc() { return false; }
+  // Is an extended descriptor needed given the element type of a fir.box type ?
+  // Extended descriptors are required for derived type and unlimited polymorphic
+  // entities.
```

Comment:
I'm not sure about the unlimited polymorphic entity bit. A boxed scalar INTEGER*4 shouldn't require an addendum (and can even elide the `dim` member information completely) as it has a unique `type` value in the degenerate descriptor, etc. I prefer passing references to degenerate types in the calls and letting the callee cast the pointer itself, rather than having a reference to an incomplete (by intention) object in memory.

Reply:
As discussed, use case is when we need to allocate descriptor for unlimited polymorphic that might later be allocated to derived type. I was assuming it would have `fir.box<None>`, but maybe another type would be better. I reverted fir.box<None> codegen change. We will take it when working on polymorphic entities.